### PR TITLE
Make HexEdit::paintEvent() 2x faster

### PR DESCRIPTION
**Pull request by mkow**
Changes:
* `painter.drawText()` -> `painter.drawStaticText()` + text cache (20% speedup)
* Removed unused `QSettings settings;` in a frequently called function (6% speedup)
* Changed one `i`-based loop to foreach (1.5%)
* Many other small changes, each giving few percents.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,6 @@ CMakeLists.txt.user
 # Visual Studio files
 .vs
 CMakeSettings.json
+*.vsp
+*.vspx
+*.psess
```
Comment (wiktorn) [inline .gitignore]: Doesn't belong here

File: CMakeLists.txt [hunk 2]
```diff
@@ -39,10 +39,13 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
 set(CMAKE_INSTALL_RPATH ".")
 if(CMAKE_COMPILER_IS_GNUCXX)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall --std=c++14") # All warnings
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++14") # All warnings
```
Comment (wiktorn) [inline CMakeLists.txt]: this too

File: src/ui/hexedit.cc [hunk 3]
```diff
@@ -114,8 +119,10 @@ HexEdit::HexEdit(FileBlobModel *dataModel, QItemSelectionModel *selectionModel,
       current_area_(WindowArea::HEX),
       cursor_pos_in_byte_(0),
       cursor_visible_(false),
+      selectionColor_(viewport()->palette().color(QPalette::Highlight)),
```
Comment (wiktorn) [inline src/ui/hexedit.cc]: clang-tidy: `field 'cursor_visible_' will be initialized after field 'selectionColor_'` ?

Comment (catsuryuu) [inline src/ui/hexedit.cc]: The issue is that order of initialization is different than order of fields in class.

**Comment by mkow** [comment]
Fixed, please check again.

**Pull request merged**